File: library-core/src/main/java/no/wtw/android/restserviceutils/resource/JsonEncodedQuery.java
package no.wtw.android.restserviceutils.resource;

import org.springframework.util.support.Base64;

import java.io.Serializable;

import no.wtw.android.restserviceutils.GsonSingleton;

public abstract class JsonEncodedQuery implements Serializable {

    private static final String TAG = JsonEncodedQuery.class.getSimpleName();

    @Override
    public String toString() {
        return encode(true);
    }

    protected String encode(boolean isBase64Encoded) {
        String json = GsonSingleton.getInstance().toJson(this);
        System.out.println(json);
        if (isBase64Encoded)
            return Base64.encodeBytes(json.getBytes());
        return json;
    }

}

File: library-core/src/main/java/no/wtw/android/restserviceutils/resource/JsonEncodedQuery.kt
package no.wtw.android.restserviceutils.resource

import no.wtw.android.restserviceutils.GsonSingleton
import java.io.Serializable

abstract class JsonEncodedQuery : Serializable {

    override fun toString(): String {
        return encode(true)
    }

    fun encode(isBase64Encoded: Boolean): String {
        val json = GsonSingleton.getInstance().toJson(this)
        println(json)
        return if (isBase64Encoded)
            TODO("Base64 implementation needed") // Base64.encodeBytes(json.toByteArray())
        else
            json
    }

}